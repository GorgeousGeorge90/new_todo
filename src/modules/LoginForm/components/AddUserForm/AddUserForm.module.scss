

.add_form_content {
  @apply flex flex-col items-center gap-0.5 w-[150px] text-sm;

  .add_form_input {
    @apply cursor-pointer border rounded-md hover:bg-violet-400 hover:ring-1 p-1 self-stretch;
  }

  .add_form_registration {
    @apply text-neutral-300 bg-violet-700 first-letter:uppercase self-stretch rounded-md py-0.5;
  }
}