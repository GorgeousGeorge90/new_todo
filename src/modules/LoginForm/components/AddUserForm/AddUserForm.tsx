import { SubmitHandler, useForm } from 'react-hook-form';
import { AddUserFormType, FormPropsType } from '../../types/types';
import styles from './AddUserForm.module.scss';
import React from 'react';
import { useAppDispatch } from "../../../../store/hooks";
import { addUser } from '../../store/loginSlice';
import {useNavigate} from "react-router-dom";


const AddUserForm = ({onClick}:FormPropsType) => {
    const dispatch = useAppDispatch()
    const navigate = useNavigate()
    const { register, reset, formState:{errors}, handleSubmit } = useForm<AddUserFormType>()

    const onSubmit:SubmitHandler<AddUserFormType> = data => {
        dispatch(addUser(data))
        onClick()
    }
    const backMove = (event:React.SyntheticEvent) => {
        event.preventDefault()
        onClick()
    }

    return (<form className={styles.add_form_content}
                  onSubmit={handleSubmit(onSubmit)}>
        <input className={styles.add_form_input}
               type='text'
               placeholder={'new name'}
               {...register('new_name')}
        />
        <input className={styles.add_form_input}
               type='text'
               placeholder={'new password'}
               {...register('new_password')}/>
        <button onClick={backMove}
                className={styles.add_form_back}>back</button>
        <button className={styles.add_form_registration}>registration</button>
    </form>)
}

export default AddUserForm