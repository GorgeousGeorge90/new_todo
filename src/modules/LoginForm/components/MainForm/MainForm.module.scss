

.main_form_content {
    @apply flex flex-col items-center w-[150px] text-sm;

    .main_form_label {
        @apply text-xs first-letter:uppercase self-start;
    }

    .main_form_input {
        @apply cursor-pointer border rounded-md hover:bg-violet-400 hover:ring-1 p-1 self-stretch;
    }

    .main_form_new_user {
        @apply text-xs text-red-600 first-letter:uppercase py-2 hover:scale-110;
    }

    .main_form_login {
        @apply text-neutral-300 bg-violet-700 first-letter:uppercase self-stretch rounded-md py-0.5;
    }
}