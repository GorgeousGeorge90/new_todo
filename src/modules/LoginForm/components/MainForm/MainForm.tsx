import { SubmitHandler, useForm } from 'react-hook-form';
import {FormPropsType, MainFormType} from '../../types/types';
import styles from './MainForm.module.scss';
import React from 'react';
import { useAppDispatch } from "../../../../store/hooks";
import { logIn } from "../../store/loginSlice";



const MainForm = ({onClick}:FormPropsType) => {
    const dispatch = useAppDispatch()
    const { register, reset, formState:{errors}, handleSubmit } = useForm<MainFormType>()
    const onSubmit:SubmitHandler<MainFormType> = data => {
        dispatch(logIn(data))
        reset()
    }

    const handleClick = (event:React.SyntheticEvent) => {
        event.preventDefault()
        onClick()
    }

    return (<form className={styles.main_form_content}
                  onSubmit={handleSubmit(onSubmit)}>
        <label className={styles.main_form_label}
               htmlFor='name'>name</label>
        <input className={styles.main_form_input}
               type='text'
               placeholder={'name'}
               {...register('name')}
        />
        <br/>
        <label className={styles.main_form_label}
               htmlFor='password'>password</label>
        <input className={styles.main_form_input}
               type='text'
               placeholder={'enter your password'}
               {...register('password')}/>
        <button className={styles.main_form_new_user}
                onClick={onClick}
        >don't have account yet</button>
        <button className={styles.main_form_login}>sign in</button>
    </form>)
}

export default MainForm