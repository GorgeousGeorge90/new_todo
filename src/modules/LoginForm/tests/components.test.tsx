import { render } from '@testing-library/react';
import AddUserForm from "../components/AddUserForm/AddUserForm";
import { BrowserRouter } from 'react-router-dom';
import MainForm from "../components/MainForm/MainForm";

jest.mock('react-redux')

describe('LoginForm render components', ()=> {
    it('goal: render AddUserForm',()=> {

        const onClick = jest.fn()

        const view = render(<BrowserRouter>
            <AddUserForm onClick={onClick}/>
                </BrowserRouter>)
        expect(view).toMatchSnapshot()
    })

    it('goal: render MainForm',()=> {

        const onClick = jest.fn()

        const view = render(<BrowserRouter>
            <MainForm onClick={onClick}/>
        </BrowserRouter>)
        expect(view).toMatchSnapshot()
    })

})