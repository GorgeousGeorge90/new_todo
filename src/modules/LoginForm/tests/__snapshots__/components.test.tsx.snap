// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoginForm render components goal: render AddUserForm 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <form
        class="add_form_content"
      >
        <input
          class="add_form_input"
          name="new_name"
          placeholder="new name"
          type="text"
        />
        <input
          class="add_form_input"
          name="new_password"
          placeholder="new password"
          type="text"
        />
        <button
          class="add_form_back"
        >
          back
        </button>
        <button
          class="add_form_registration"
        >
          registration
        </button>
      </form>
    </div>
  </body>,
  "container": <div>
    <form
      class="add_form_content"
    >
      <input
        class="add_form_input"
        name="new_name"
        placeholder="new name"
        type="text"
      />
      <input
        class="add_form_input"
        name="new_password"
        placeholder="new password"
        type="text"
      />
      <button
        class="add_form_back"
      >
        back
      </button>
      <button
        class="add_form_registration"
      >
        registration
      </button>
    </form>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`LoginForm render components goal: render MainForm 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <form
        class="main_form_content"
      >
        <label
          class="main_form_label"
          for="name"
        >
          name
        </label>
        <input
          class="main_form_input"
          name="name"
          placeholder="name"
          type="text"
        />
        <br />
        <label
          class="main_form_label"
          for="password"
        >
          password
        </label>
        <input
          class="main_form_input"
          name="password"
          placeholder="enter your password"
          type="text"
        />
        <button
          class="main_form_new_user"
        >
          don't have account yet
        </button>
        <button
          class="main_form_login"
        >
          sign in
        </button>
      </form>
    </div>
  </body>,
  "container": <div>
    <form
      class="main_form_content"
    >
      <label
        class="main_form_label"
        for="name"
      >
        name
      </label>
      <input
        class="main_form_input"
        name="name"
        placeholder="name"
        type="text"
      />
      <br />
      <label
        class="main_form_label"
        for="password"
      >
        password
      </label>
      <input
        class="main_form_input"
        name="password"
        placeholder="enter your password"
        type="text"
      />
      <button
        class="main_form_new_user"
      >
        don't have account yet
      </button>
      <button
        class="main_form_login"
      >
        sign in
      </button>
    </form>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
